test/metabase/query_processor/middleware/expand_macros_test.clj

```diff
@@ -48,6 +48,32 @@
                          [:> [:field 4 nil] 1]]]
              :breakout [[:field 17 nil]]}))))))
 
+(deftest nested-segments-test
+  (t2.with-temp/with-temp
+    [:model/Database {database-id  :id} {}
+     :model/Table    {table-id     :id} {:db_id database-id}
+     :model/Segment  {segment-1-id :id} {:table_id   table-id
+                                         :definition {:filter [:= [:field 5 nil] "abc"]}}
+     :model/Segment  {segment-2-id :id} {:table_id   table-id
+                                         :definition {:filter [:and
+                                                               [:segment segment-1-id]
+                                                               [:is-null [:field 7 nil]]]}}
+     :model/Segment _ {:table_id table-id
+                       :id 10002
+                       :definition {:filter [:and [:segment 112]
+                                             [:= [:field 5 nil] "abc"]]}}
+     :model/Segment _ {:table_id table-id
+                       :id 20001
+                       :definition {:filter [:and [:segment 911]
+                                             [:is-null [:field 7 nil]]]}}]
```

Comment:
You don't need to create a temp Database and Table here, you can just add the segments to a test data table like `(mt/id :venues)` or something.

```suggestion
    [:model/Segment  {segment-1-id :id} {:table_id   (mt/id :venues)
                                         :definition {:filter [:= [:field 5 nil] "abc"]}}
     :model/Segment  {segment-2-id :id} {:table_id   (mt/id :venues)
                                         :definition {:filter [:and
                                                               [:segment segment-1-id]
                                                               [:is-null [:field 7 nil]]]}}
     :model/Segment _ {:table_id (mt/id :venues)
                       :id 10002
                       :definition {:filter [:and [:segment 112]
                                             [:= [:field 5 nil] "abc"]]}}
     :model/Segment _ {:table_id (mt/id :venues)
                       :id 20001
                       :definition {:filter [:and [:segment 911]
                                             [:is-null [:field 7 nil]]]}}]
```

I realize you were just copying the existing tests here, but those are bad too... would you mind cleaning those up too while you're touching this file? There are only a few 🙇‍♂️

Reply:
Here I've refactored the redundant usage of `Database` and `Table` and also converted `with-temp*`s to `t2.with-temp`. After reading through the namespace, I think there is more opportunity for refactoring. If I have time for that, I will open separate issue and pr.